dvc/repo/metrics/show.py

```diff
@@ -55,7 +68,7 @@ def _read_metrics(repo, metrics, rev):
 
     res = {}
     for metric in metrics:
-        if not tree.exists(metric):
```

Comment:
Supporting `dir plots` allows us to `target` dirs during `collect`. Since we also support `target`-ing non-metrics, it is possible that someone will run `dvc metrics show dir`. In that situation, we would try to parse dir with yaml parser. I am narrowing this check, to avoid such a situation.